src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -227,6 +239,37 @@ internal static int GetRootLength(string path)
             return i;
         }
 
+        /// <summary>
+        /// Returns true if the path specified is relative to the current drive or working directory.
+        /// Returns false if the path is fixed to a specific drive or UNC path.  This method does no
+        /// validation of the path (URIs will be returned as relative as a result).
+        /// </summary>
+        /// <remarks>
+        /// Handles paths that use the alternate directory separator.  It is a frequent mistake to
+        /// assume that rooted paths (Path.IsPathRooted) are not relative.  This isn't the case.
```

Comment:
> I'll call the "C:" case out explicitly

Thanks

> The Unix impl would be "starts with \"?

You mean "not starts with /", right?  In other words, it's relative iff it's not rooted.